src/drivers/event/mod.rs

```diff
@@ -38,8 +38,15 @@ pub async fn subscribe(config: EventConfig) -> Result<()> {
         match consumer.recv().await {
             Err(err) => error!(error = err.to_string(), "kafka subscribe error"),
             Ok(message) => {
+                let Some(key) = message.key() else {
+                    warn!("event with empety key");
+                    consumer.commit_message(&message, CommitMode::Async)?;
+                    continue;
+                };
+                let key = String::from_utf8(key.to_vec())?;
+
                 if let Some(payload) = message.payload() {
-                    let event: Event = serde_json::from_slice(payload)?;
+                    let event = Event::from_key(&key, payload)?;
```

Comment:
What do you think about doing a `Event::from(&message)` directly? To convert the kafka message into an even directly. I mean implementing
```rust

impl From<&rdkafka::message::BorrowedMessage> for Event {
    fn from ...
}

Reply:
yes, it's better